web/core/src/jsTest/kotlin/TestUtils.kt

```diff
@@ -1,68 +1,167 @@
+     */
     fun composition(content: @Composable () -> Unit) {
         root.clear()
-        renderComposable(root) {
+
+        renderComposable(
+            root = root,
+            monotonicFrameClock = TestMonotonicClockImpl(
+                onRecomposeComplete = this::onRecompositionComplete
+            )
+        ) {
             content()
         }
     }
 
-
-    private val childrenIterator = root.children.asList().listIterator()
+    /**
+     * @return a reference to the next child element of the root.
+     * Subsequent calls will return next child reference every time.
+     */
     fun nextChild() = childrenIterator.next() as HTMLElement
+
+    /**
+     * @return a reference to current child.
+     * Calling this subsequently returns the same reference every time.
+     */
     fun currentChild() = root.children[childrenIterator.previousIndex()] as HTMLElement
 
-    suspend fun waitChanges() {
+    /**
+     * Suspends until [root] observes any change to its html.
+     */
+    suspend fun waitForChanges() {
         waitForChanges(root)
     }
-}
-
-internal fun runTest(block: suspend TestScope.() -> Unit): dynamic {
-    val scope = TestScope()
-    return scope.promise { block(scope) }
-}
 
-internal fun runBlockingTest(
-    block: suspend CoroutineScope.() -> Unit
-): dynamic = testScope.promise { this.block() }
+    /**
+     * Suspends until element with [elementId] observes any change to its html.
+     */
+    suspend fun waitForChanges(elementId: String) {
+        waitForChanges(document.getElementById(elementId) as HTMLElement)
+    }
 
-internal fun String.asHtmlElement() = document.createElement(this) as HTMLElement
+    /**
+     * Suspends until [element] observes any change to its html.
+     */
+    suspend fun waitForChanges(element: HTMLElement) {
+        suspendCoroutine<Unit> { continuation ->
+            val observer = MutationObserver { mutations, observer ->
+                continuation.resume(Unit)
+                observer.disconnect()
+            }
+            observer.observe(element, MutationObserverOptions)
+        }
+    }
 
-/* Currently, the recompositionRunner relies on AnimationFrame to run the recomposition and
-applyChanges. Therefore we can use this method after updating the state and before making
-assertions.
+    /**
+     * Suspends until recomposition completes.
+     */
+    suspend fun waitForRecompositionComplete() {
+        recompositionCompleteEventsChannel.receive()
+    }
 
-If tests get broken, then DefaultMonotonicFrameClock need to be checked if it still
-uses window.requestAnimationFrame */
-internal suspend fun waitForAnimationFrame() {
-    suspendCoroutine<Unit> { continuation ->
-        window.requestAnimationFrame {
-            continuation.resume(Unit)
-        }
+    @Deprecated(
+        message = "The intention of this function was to wait for recompositions. Now there is a more specific function",
```

Comment:
My understanding is that it's only us who used this code - so do we actually need to deprecate it?

Reply:
since it's going to be public, I decided to deprecate it for us as well.
Actually, I believe there is no reason to keep this function. So I replace it's usages by  `waitForRecompositionComplete` and remove deprecated `waitForAnimationFrame `